src/dune_rules/melange/melange_rules.ml

```diff
@@ -247,6 +247,66 @@ let setup_emit_cmj_rules ~sctx ~dir ~scope ~expander ~dir_contents
   in
   Buildable_rules.with_lib_deps ctx compile_info ~dir ~f
 
+module Runtime_deps = struct
+  let to_action_builder ~expander dep_conf =
+    let runtime_deps =
+      let runtime_deps, _sandbox =
+        Dep_conf_eval.unnamed_get_paths ~expander dep_conf
+      in
+      runtime_deps
+    in
+    Action_builder.memoize "melange runtime deps" runtime_deps
+
+  let eval ~expander (deps : Dep_conf.t list) =
+    let open Memo.O in
+    let builder = to_action_builder ~expander deps in
+    let+ paths, _ = Action_builder.run builder Eager in
+    paths
+
+  let targets ~output deps =
+    Path.Set.to_list_map
+      ~f:(fun src ->
+        let dst =
+          match output with
+          | `Private_library_or_emit output_dir ->
+            let target =
+              src |> Path.as_in_build_dir_exn
```

Comment:
I believe this would fail if there are any external paths in runtime_deps. Worth adding a test for this.